### ISSUE TITLE
Store drop is not working properly

### ISSUE THREAD

**Issue by saravananmnm** [tekartik/sembast.dart#104]
await loginStore.drop(await appDatabase.database);
it was returned null.But still records is in local storage.
delete is working fine for delete all records.But auto-increment id is incremented next one. if i try to store new records to same store.
sembast: ^2.1.2

Is there any solution for entire store delete?

**Comment by saravananmnm** [comment]
[√] Flutter (Channel stable, v1.9.1+hotfix.6, on Microsoft Windows [Version 10.0.18362.535], locale en-IN)
    • Flutter version 1.9.1+hotfix.6 at C:\Saravanan\Flutter_Sdk\flutter
    • Framework revision 68587a0916 (3 months ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\Admin\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Connected device (1 available)
    • Genymotion • 192.168.34.101:5555 • android-x86 • Android 9 (API 28)

• No issues found!

**Comment by alextekartik** [comment]
Hi, I'm not sure what you mean by 'it was returned null', but indeed the method return a future and the value awaited has no meaning (in fact it should event be a Future<void>). Is the store properly deleted and where you expecting something else?

**Comment by saravananmnm** [comment]
it was returned null.But still records is in local storage.
delete is working fine for delete all records.But auto-increment id is incremented next one. if i try to store new records to same store.

**Comment by alextekartik** [comment]
> it was returned null.But still records is in local storage.

Do you have some code or unit tests to reproduce the issue, as I'm not sure what you mean by 'is in local storage'.

> .But auto-increment id is incremented next one.

Indeed id are not resetted to 0. Application should not rely on that though.

> Is there any solution for entire store delete?

drop should be the solution

**Comment by mariomka** [comment]
What is the difference between `delete` and `drop` for a store?

I found an issue when you drop a store and closing and re-opening the database.

```dart
import 'dart:io';

import 'package:sembast/sembast.dart';
import 'package:sembast/sembast_io.dart';
import 'package:test/test.dart';

void main() {
  Database db;
  String dbPath = 'test.db';

  tearDown(() async {
    await db.close();
    await File(dbPath).delete();
  });

  test('drop store', () async {
    // Open the db
    db = await databaseFactoryIo.openDatabase(dbPath);

    // Add a record to the store
    var dbStore = stringMapStoreFactory.store('store1');
    await dbStore.record('key').put(db, {'key': 'value'});

    // Drop the store
    await dbStore.drop(db);

    // Close the db
    await db.close();
    db = await databaseFactoryIo.openDatabase(dbPath);

    // Get the record
    var record = await dbStore.record('key').get(db);

    // Assert
    expect(record, null);
  });
}
```

The test pass commenting close/open the DB.

**Comment by mariomka** [comment]
Ok, `drop` should delete the whole store and its contents and `delete` should delete only its contents (or matched contents).
But it seems `drop` isn't saved correctly in the database file.

**Comment by alextekartik** [comment]
Thanks @mariomka for taking the time to write a unit test that pinpoints that issue. Indeed there were an issue with `drop`. There used to be a concept of existing stores but its behavior was not consistent so indeed now drop and delete has the same behavior (delete allows adding a filter).

I fixed the drop issue in the version 2.1.2+1.

**Comment by mariomka** [comment]
Thank you!
